backend/src/main/scala/cromwell/backend/wdl/OutputEvaluator.scala

```diff
@@ -40,9 +41,17 @@ object OutputEvaluator {
       }
 
       // Attempt to evaluate the expression using all known values
-      def evaluateOutputExpression: OutputResult[WomValue] = {
-        EitherT { Try(output.expression.evaluateValue(allKnownValues, ioFunctions)).map(_.toEither) }
-      }
+      def evaluateOutputExpression: OutputResult[WomValue] = EitherT{ Try{
+        output.expression.evaluateValue(allKnownValues, ioFunctions).toEither.map{
+
+          //If the string contained in a filename is actually a glob,
+          // The output of this expression is expected to be glob-md5(fileName).list
+          case WomSingleFile(fileName) if fileName.contains("*") =>
```

Comment:
This seems a little off to me:
- The output evaluator should be returning the final value, not some intermediate thing in need of further interpretation.
- As part of evaluation, it should notice that it's using a glob and do the stuff with the `.list` file behind the scenes
- Maybe the standard `IoFunctions` that Cromwell passes to the languages should contain a `resolveGlobOutput` method from `String` to `Array[File]`?

Reply:
`IoFunctions` has 
```scala
glob(pattern: String): Seq[String]
```
However the Cromwell implementation is expecting to see a `.list` file from this output evaluation, and it does the dereferencing downstream of this output

Reply:
Totally not your fault but if that's the case then I would tend to say it shouldn't be. The whole point of `IoFunctions` is that they are provided by the engine which gives it full flexibility as to how it wants to perform them, and if that includes dereferencing things it should be done as part of this method IMO.